### PR TITLE
Optional colour map for heatmap data

### LINKED ISSUE #166: Make colour map optional in `HeatmapData`
Currently using the `image` function in `plot.py` requires the user to supply a `colour_map` argument.

1. make this optional and use the colour map of any heatmap plotted already or a default, perhaps, "Gray"
2. `colour_map` is a `str` in `HeatmapData`. Make it an enum to ensure it exists on the client.

### PR DESCRIPTION
**Pull request by VictoriaBeilsten-Edmands**
`heatmap_data`'s `colour_map` is `Optional[ColourMap | Literal["Last"]]` and defaults to `default_colour_map` if `None`. `default_colour_map` is initialised to `ColourMap.Greys` and `last_colour_maps` is updated whenever a heatmap with `colour_map` is plotted.
Closes #166

### CODE REVIEW

File: server/davidia/models/messages.py [hunk 1]
```diff
@@ -56,6 +56,48 @@ class GlyphType(str, Enum):
     Cap = "Cap"
 
 
+class ColourMap(str, Enum):
+    Blues = "Blues"
```
Comment (PeterC-DLS) [inline server/davidia/models/messages.py:60]: Can define an enum subtype with [_generate_next_value_](https://docs.python.org/3/library/enum.html#enum.Enum._generate_next_value_) and use `auto()`

File: server/davidia/models/messages.py [hunk 2]
```diff
@@ -56,6 +56,48 @@ class GlyphType(str, Enum):
...
+    BrBG = "BrBG"
+    PRGn = "PRGn"
+    PiYG = "PiYG"
+    PuOr = "PuOr"
+    RdBu = "RdBu"
+    RdGy = "RdGy"
+    RdYlBu = "RdYlBu"
+    RdYlGn = "RdYlGn"
+    Spectral = "Spectral"
+
```
Comment (PeterC-DLS) [inline server/davidia/models/messages.py:99]: Add `Last` here

File: server/davidia/models/messages.py [hunk 3]
```diff
@@ -136,7 +178,7 @@ class HeatmapData(ImageData):
 
     domain: tuple[float, float]
     heatmap_scale: ScaleType = ScaleType.linear
-    colour_map: str
+    colour_map: Optional[ColourMap | Literal["Last"]] = None
```
Comment (PeterC-DLS) [inline server/davidia/models/messages.py:181]: Add pydoc to explain semantics

File: server/davidia/models/messages.py [hunk 4]
```diff
@@ -1,5 +1,5 @@
-from enum import Enum
-from typing import Any
+from enum import auto, Enum
+from typing import Any, Literal, Optional
```
Comment (PeterC-DLS) [inline server/davidia/models/messages.py:2]: Don't need `Literal` anymore

File: server/davidia/models/messages.py (hunk 1 above)
Comment (VictoriaBeilsten-Edmands) [reply server/davidia/models/messages.py]: Added `_generate_next_value()` static method and `auto()`.
Could use a list e.g. `["Blues", "Greens", etc]` and `setattr()`  to loop through instead to cut down "Colour = auto()" code.

File: server/davidia/models/messages.py (hunk 2 above)
Comment (VictoriaBeilsten-Edmands) [reply server/davidia/models/messages.py:104]: Added

File: server/davidia/models/messages.py (hunk 3 above)
Comment (VictoriaBeilsten-Edmands) [reply server/davidia/models/messages.py]: I've removed `Literal["Last"]` as it's now in ColourMap

File: server/davidia/models/messages.py [hunk 5]
```diff
@@ -136,7 +183,7 @@ class HeatmapData(ImageData):
 
     domain: tuple[float, float]
     heatmap_scale: ScaleType = ScaleType.linear
-    colour_map: str
+    colour_map: Optional[ColourMap] = None
```
Comment (PeterC-DLS) [inline server/davidia/models/messages.py:186]: docstring should go after attribute

File: server/davidia/models/messages.py [hunk 6]
```diff
@@ -136,7 +183,8 @@ class HeatmapData(ImageData):
...
+    "An optional parameter which can also be set to ColourMap.Last to use the last provided ColourMap colour"
```
Comment (PeterC-DLS) [inline server/davidia/models/messages.py:187]: "... and if None, use the default colour map defined in the plot server"

**Review by PeterC-DLS — APPROVED** [review]

**Pull request merged**